src/portals/hope/common/pcms/usePcms.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useCallback } from 'react'
+import axios from 'axios'
+import { Locales } from 'portals/hope/features/config/constants'
+import { createContext, useContext, useEffect, useState } from 'react'
+
+export interface CommonClaim {
+  id: number
+  attributes: {
+    title: string
+    locale: string
+    type: string
+  }
+  locale: Locales
+}
+
+const PcmsContext = createContext<{
+  allCommonClaims: CommonClaim[]
+  getCommonClaim: (commonClaimId: number) => CommonClaim | undefined
+}>({
+  allCommonClaims: [],
+  getCommonClaim: () => undefined,
+})
+
+export const usePcms = () => useContext(PcmsContext)
+
+export const PcmsProvider: React.FC = ({ children }) => {
+  const [allCommonClaims, setAllCommonClaims] = useState<CommonClaim[]>([])
+  const getCommonClaims = async (locale: string): Promise<CommonClaim[]> => {
+    return (
+      await axios.get(`/pcms/common-claims?locale=${locale}&populate=*`)
+    )?.['data']?.['data'] as CommonClaim[]
+  }
+
+  const getAllCommonClaims = useCallback(async () => {
+    return Promise.all(
+      Object.values(Locales).map((locale) => getCommonClaims(locale)),
+    ).then((commonClaimLists) => {
+      return commonClaimLists.flat().sort((a, b) => +a.id - +b.id)
+    })
+  }, [])
+
+  useEffect(() => {
+    const setupAllCommonClaims = async () => {
+      getAllCommonClaims().then((all) => setAllCommonClaims(all))
+    }
+    setupAllCommonClaims()
```

Comment:
Do you really need the extra function? Feel like this should work
```suggestion
    getAllCommonClaims().then(setAllCommonClaims)
```